twenty-first/src/math/bfield_codec.rs

```diff
@@ -373,6 +378,39 @@ impl<T: BFieldCodec> BFieldCodec for Vec<T> {
     }
 }
 
+impl<T: BFieldCodec + FiniteField> BFieldCodec for Polynomial<T> {
+    type Error = BFieldCodecError;
+
+    fn decode(sequence: &[BFieldElement]) -> Result<Box<Self>, Self::Error> {
+        let decoded_vec = Vec::<T>::decode(sequence)?;
+
+        let contains_trailing_zeros = match decoded_vec.last() {
+            Some(last_coeff) => last_coeff.is_zero(),
+            None => false,
+        };
+
+        if contains_trailing_zeros {
```

Comment:
More canonical rust:

```suggestion
        if decoded_vec.last().is_some_and(|c| c.is_zero()) {
```